backend/molgenis-emx2-sql/src/main/java/org/molgenis/emx2/sql/SqlTypeUtils.java

```diff
@@ -89,21 +89,25 @@ private static void applyAutoid(Column c, Row row) {
     }
   }
 
-  private static void checkRequired(Column c, Row row) {
-    if (!row.isDraft()
-        && c.getComputed() == null
-        && !AUTO_ID.equals(c.getColumnType())
-        && c.isRequired()) {
-
-      if (c.isReference()) {
-        for (Reference r : c.getReferences()) {
-          if (row.isNull(r.getName(), r.getPrimitiveType())) {
+  private static void checkRequired(Column c, Row row, Map<String, Object> values) {
+    if (!row.isDraft() && c.getComputed() == null && !AUTO_ID.equals(c.getColumnType())) {
+      if (c.isRequired()) {
```

Comment:
I think this is still incorrect I believe. I think it should be one 'if' without your new else that looks something like

`if(c.isRequired || executeRequiredExpression(c.getExpression)) { 
... //here the original check code to see if the values are not null 
}`

with something like
`boolean executeRequiredExpression(Column c, Map values) {
  String result = checkValidation(c.getRequired(), values);
  return result != null && !"false".equals(result)
}
`